numba/np/polynomial.py

```diff
@@ -57,3 +58,150 @@ def roots_impl(p):
             return roots
 
     return roots_impl
+
+
+@overload(pu.trimseq)
+def polyutils_trimseq(seq):
+    if not type_can_asarray(seq):
+        msg = 'The argument "seq" must be array-like'
+        raise errors.TypingError(msg)
+
+    if np.ndim(seq) > 1:
+        msg = 'Coefficient array is not 1-d'
+        raise errors.NumbaValueError(msg)
+    
+    def impl(seq):
+        if len(seq) == 0:
+            return seq
+        else:
+            for i in range(len(seq) - 1, -1, -1):
+                if seq[i] != 0:
+                    break
+            return seq[:i+1]
+    
+    return impl
+
+
+@overload(poly.polyadd)
+def numpy_polyadd(c1, c2):
+    if not type_can_asarray(c1):
+        msg = 'The argument "c1" must be array-like'
+        raise errors.TypingError(msg)
+    
+    if not type_can_asarray(c2):
+        msg = 'The argument "c2" must be array-like'
+        raise errors.TypingError(msg)
+
+    if np.ndim(c1) > 1 or np.ndim(c2) > 1:
+        msg = 'Coefficient array is not 1-d'
+        raise errors.NumbaValueError(msg)
+
+    if isinstance(c1, types.Integer):
+        s1 = str(as_dtype(typeof(c1)))
+    else:
+        s1 = str(c1.dtype)
+
+    if isinstance(c2, types.Integer):
+        s2 = str(as_dtype(typeof(c2)))
+    else:
+        s2 = str(c2.dtype)
+
+    result_dtype = from_dtype(np.result_type(s1, s2, np.float64))
```

Comment:
The last argument in `np.result_type` will force the promotion to float64:

```
>>> a = np.arange(5, dtype=np.int32)
>>> b = np.arange(5, dtype=np.int16)
>>> np.polyadd(a, b)
array([0, 2, 4, 6, 8], dtype=int32)
```

Reply:
Hi @guilhermeleobas, I believe you are using the old API. The new ``np.polynomial`` package promotes to float64:
```
>>> import numpy as np
>>> import numpy.polynomial.polynomial as poly
>>> a = np.arange(5, dtype=np.int32)
>>> b = np.arange(5, dtype=np.int16)
>>> c = poly.polyadd(a, b)
>>> c
array([0., 2., 4., 6., 8.])
>>> c.dtype
dtype('float64')
```